internal/coreinternal/attraction/attraction.go

```diff
@@ -311,8 +311,38 @@ func (ap *AttrProc) Process(ctx context.Context, logger *zap.Logger, attrs pcomm
 }
 
 func getAttributeValueFromContext(ctx context.Context, key string) (pcommon.Value, bool) {
+	const (
+		metadataPrefix = "metadata."
+		authPrefix     = "auth."
+	)
+
 	ci := client.FromContext(ctx)
-	vals := ci.Metadata.Get(key)
+	var vals []string
+
+	switch {
+	case strings.HasPrefix(key, metadataPrefix):
+		mdKey := strings.TrimPrefix(key, metadataPrefix)
+		vals = ci.Metadata.Get(mdKey)
+	case strings.HasPrefix(key, authPrefix):
+		if ci.Auth == nil {
+			return pcommon.Value{}, false
+		}
+
+		attrName := strings.TrimPrefix(key, authPrefix)
+		attr := ci.Auth.GetAttribute(attrName)
+
+		switch a := attr.(type) {
+		case string:
+			return pcommon.NewValueString(a), true
+		case []string:
+			vals = a
+		default:
+			return pcommon.Value{}, false
```

Comment:
What to do with non-string typed attributes? At the moment we don't have any use cases as auth extensions either return `string` or `[]string` attributes.

Should I match on raw type and produce corresponding value? I would probably do that if `newValueFromRaw` was public. https://github.com/open-telemetry/opentelemetry-collector/blob/6fed15bf12d807da8ac834fd776eb1423187ca19/pdata/internal/common.go#L135

Reply:
I wonder if this might ever happen. Perhaps we should just log that unexpected type was observed (ideally only once for a while, e.g. using logging sampler or other means)

Reply:
I consider this an error condition and users should be appropriately warned if this ever happens.

Reply:
Added log warning.